js/renovation/icon.tsx

```diff
@@ -0,0 +1,55 @@
+import { getImageSourceType } from '../core/utils/icon';
+import { Component, ComponentBindings, JSXComponent, OneWay, Fragment } from 'devextreme-generator/component_declaration/common';
+
+type IconViewModel = {
+    cssClass: string;
+    props: IconInput;
+    sourceType: string;
+};
+
+export const viewModelFunction = (model: Icon): IconViewModel => {
+    return {
+        cssClass: model.props.position !== 'left' ? 'dx-icon-right' : '',
+        props: model.props,
+        sourceType: getImageSourceType(model.props.source),
+    };
+};
+
+export const viewFunction = (viewModel: IconViewModel) => {
+    return (<Fragment>
```

Comment:
Should we decompose `viewModel`?
Something like
```tsx
export const viewFunction = ({ sourceType, cssClass, props: { source } }: IconViewModel) => {
```

Reply:
I like it

Reply:
I agreed with Andrey. We often use objects decomposition.